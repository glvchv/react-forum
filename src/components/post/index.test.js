import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Post from '.';
const props = {
    post: {
        _id: '123',
        text: 'Testing the description value of the post',
        title: 'TESTTESTTEST',
        category: 'other',
        likes: [1,1,1,1],
        replies: [1,1,1,1],
        date: '13.09.2020',
        author: {
            _id: '123',
            avatarUrl: 'https://localhost:0000/test',
            username: 'vld',
        },
        hasLiked: false,
        handleClick: () => {}
    }
}
describe('Post component', () => {
    it('should render not liked post', () => {
        const tree = renderer.create(
            <BrowserRouter>
                <Post {...props}/>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render liked post', () => {
        props.hasLiked= true;
        const tree = renderer.create(
            <BrowserRouter>
                <Post {...props}/>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render post for author', () => {
        localStorage.setItem('userId', '123');
        const tree = renderer.create(
            <BrowserRouter>
                <Post {...props}/>
            </BrowserRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})