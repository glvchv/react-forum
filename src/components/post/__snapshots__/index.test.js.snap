// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render liked post 1`] = `
<div
  className="post-body"
>
  <div
    className="post-details"
  >
    <img
      alt="avatar"
      className="avatar"
      src="https://localhost:0000/test"
    />
    <p>
      <a
        href="/profile/123"
        onClick={[Function]}
      >
        vld
      </a>
    </p>
    <small>
      Invalid Date
    </small>
  </div>
  <div
    className="main"
  >
    <h2
      className="title"
    >
      TESTTESTTEST
    </h2>
    <p
      className="text"
    >
      Testing the description value of the post
    </p>
    <hr />
    <div
      className="actions"
    >
      <p>
         
        <small>
          likes
        </small>
      </p>
      <p>
        <small>
          Category: 
        </small>
        other
      </p>
      <p>
        4
         
        <small>
          replies
        </small>
      </p>
      <span
        className="like"
      >
        <button
          disabled={true}
        >
          ðŸ’™
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`Post component should render not liked post 1`] = `
<div
  className="post-body"
>
  <div
    className="post-details"
  >
    <img
      alt="avatar"
      className="avatar"
      src="https://localhost:0000/test"
    />
    <p>
      <a
        href="/profile/123"
        onClick={[Function]}
      >
        vld
      </a>
    </p>
    <small>
      Invalid Date
    </small>
  </div>
  <div
    className="main"
  >
    <h2
      className="title"
    >
      TESTTESTTEST
    </h2>
    <p
      className="text"
    >
      Testing the description value of the post
    </p>
    <hr />
    <div
      className="actions"
    >
      <p>
         
        <small>
          likes
        </small>
      </p>
      <p>
        <small>
          Category: 
        </small>
        other
      </p>
      <p>
        4
         
        <small>
          replies
        </small>
      </p>
      <span
        className="like"
      >
        <button>
          ðŸ’™
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`Post component should render post for author 1`] = `
<div
  className="post-body"
>
  <div
    className="post-details"
  >
    <img
      alt="avatar"
      className="avatar"
      src="https://localhost:0000/test"
    />
    <p>
      <a
        href="/profile/123"
        onClick={[Function]}
      >
        vld
      </a>
    </p>
    <small>
      Invalid Date
    </small>
  </div>
  <div
    className="main"
  >
    <h2
      className="title"
    >
      TESTTESTTEST
    </h2>
    <p
      className="text"
    >
      Testing the description value of the post
    </p>
    <hr />
    <div
      className="actions"
    >
      <p>
         
        <small>
          likes
        </small>
      </p>
      <p>
        <small>
          Category: 
        </small>
        other
      </p>
      <p>
        4
         
        <small>
          replies
        </small>
      </p>
      <span
        className="buttons"
      >
        <button
          className="delete"
          onClick={[Function]}
        >
          Delete
        </button>
      </span>
    </div>
  </div>
</div>
`;
